src/pages/_app.tsx

```diff
@@ -44,8 +44,14 @@ const App = ({
   const [isRouteChangeInProgress, setIsRouteChangeInProgress] = useState(false)
   const { colorMode } = useColorMode()
 
+  let previousPathname = null
```

Comment:
Shouldn't we use useRef or useState here? I haven't tried if it works yet, but I think we should since we're in a component and they also mention it in a comment on Stackoverflow: https://stackoverflow.com/questions/62368109/update-router-query-without-firing-page-change-event-in-next-js#comment119198796_62368514